src/basic-samples/ConfirmedPublish.js

```diff
@@ -149,9 +141,43 @@ var QueueProducer = function (solaceModule, queueName) {
             id: sequenceNr,
         };
         message.setCorrelationKey(correlationKey);
+        return message; // return the message
+    };
+
+    // Send many message
+    producer.sendMessages = function () {
+        if (producer.session !== null) {
+            var sentCount = 0;
+            const doSend = () => {
+                try {
+                    producer.log(`Starting send at: ${sentCount}`);
+                    while (sentCount < producer.numOfMessages) {
+                        var sequenceNr = sentCount + 1;
+                        var message = producer.getMessage(sequenceNr);
+                        producer.session.send(message);
+                        producer.log('Message #' + sequenceNr + ' sent to queue "' + producer.queueName + '", correlation key = ' + JSON.stringify(message.getCorrelationKey()));
+                        ++sentCount;
+                    }
+                } catch (error) {
+                    producer.log(`Aborting send at: ${sentCount}`);
+                    producer.log(error.toString());
+                    producer.session.once(solace.SessionEventCode.CAN_ACCEPT_DATA, () => {
+                        doSend();
+                    });
+                }
+              };
+              doSend();
+        } else {
+            producer.log('Cannot send messages because not connected to Solace PubSub+ Event Broker.');
+        }
+    }
+
+    // Sends one message
+    producer.sendMessage = function (sequenceNr) {
```

Comment:
We dont really use this function anywhere correct? Might be worth adding a comment that this is not used but keep it for reference

Reply:
Yes. The original code didn't use it as well. I will ass the comment. Thanks